proposals/extensions.md

```diff
@@ -398,23 +390,27 @@ static class IEnumerableExtensions
     }
 
     // Implementation for Method
-    public static void <Extension>Method<T>(IEnumerable<T> source) { ... }
+    [Extension]
+    public static void Method<T>(IEnumerable<T> source) { ... }
 
     // Implementation for Property
-    internal static int <StaticExtension>get_Property<T>() { ... }
-    internal static void <StaticExtension>set_Property<T>(int value) { ... }
+    internal static modopt(IEnumerable<T>) int get_Property<T>() { ... }
+    internal static modopt(IEnumerable<T>) void set_Property<T>(int value) { ... }
 
     // Implementation for SumAsync
-    public static int <Extension>SumAsync(IAsyncEnumerable<int> values) { ... }
+    public static int SumAsync(IAsyncEnumerable<int> values) { ... }
+
+    public void Method2() { ... }
 }
 ```
 
 Whenever extension members are used in source, we will emit those as reference to implementation methods.  
 For example: an invocation of `enumerableOfInt.Method()` would be emitted as a static call 
-to `IEnumerableExtensions.<Extension>Method<int>(enumerableOfInt)`.  
+to `IEnumerableExtensions.Method<int>(enumerableOfInt)`.  
 
-Note: multiple extension declarations defining static members with the same signature cannot be represented in metadata.  
-However, differences in return types can be represented, allowing for factory methods extending different types. For example:
+Note: the metadata representation supports static extension methods that differ in receiver type and return type, but
+  this will be disallowed in the language.
```

Comment:
Might want to double check LDM notes for this. At least I do not remember us making an explicit decision like that.